File: 0148-sort-list/0148-sort-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def sortList(self, head):
        li=[]
        current=head
        while current:
            li.append(current.val)
            current=current.next

        if not li:
            return None    

        li.sort()
        newhead=ListNode(li[0])
        current=newhead
        for i in range(1,len(li)):
            current.next=ListNode(li[i])
            current=current.next

        return newhead      

File: 0148-sort-list/0148-sort-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* sortList(ListNode* head) {
        vector <int> arr;
        ListNode* current=head;
        while(current!=NULL){
            arr.push_back(current->val);
            current=current->next;
        }

        sort(arr.begin(),arr.end());
        if (arr.empty()){
            return NULL;
        }
        ListNode* new_head=new ListNode(arr[0]);
        current=new_head;

        for(int i=1;i<arr.size();i++){
            current->next=new ListNode(arr[i]);
            current=current->next;

        }

        return new_head;
        
    }
};